sockeye/inference.py

```diff
@@ -159,9 +160,15 @@ def sym_gen(source_seq_len: int):
             source_length = utils.compute_lengths(source)
 
             # source embedding
-            (source_embed,
-             source_embed_length,
-             source_embed_seq_len) = self.embedding_source.encode(source, source_length, source_seq_len)
+            if self.num_factors == 0:
+                (source_embed,
+                 source_embed_length,
+                 source_embed_seq_len) = self.embedding_source.encode(source, source_length, source_seq_len)
+            else:
+                source_factors = [mx.sym.Variable(C.SOURCE_NAME + "_factor_" + str(i)) for i in range(self.config.num_factors)]
+                (source_embed,
+                 source_embed_length,
+                 source_embed_seq_len) = self.embedding_source.encode(source, source_length, source_seq_len, source_factors)
```

Comment:
See note below for `training.py`: it might be better to encode factors as additional dimensions on the `source` ndarray, which `FactoredEmbedding` would know to look for. We could also do away with `FactoredEmbedding` entirely, and have the embedding receive the number of factors when it is instantiated.

Reply:
I think it might be a good idea to add the factor functionality to the existing Embedding class (via arguemnts in its constructor)